dlfcn.c

```diff
@@ -474,6 +474,48 @@ char *dlerror( void )
     return error_buffer;
 }
 
+int dladdr (const void *addr, Dl_info *info)
+{
+   // only returns filename, FWIW.
+   TCHAR  tpath[PATH_MAX];
+   MEMORY_BASIC_INFORMATION mbi;
+   char  *path;
+   char  *tmp;
+   size_t length;
+   int    ret = 0;
+
+   if (!info)
+     return 0;
+
+   ret = GetModuleFileName(NULL, (LPTSTR)&tpath, PATH_MAX);

```

Comment:
`GetModuleFileName` is unsafe as it may expand to `GetModuleFileNameW` which returns `wchar_t` and which is incompatible with types in `struct dl_info`.